uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/ContactActionsMenu/WorkflowActionGroup.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+use CRM_HRContactActionsMenu_Component_Group as ActionsGroup;
+use CRM_HRContactActionsMenu_Component_GroupButtonItem as ActionsGroupButtonItem;
+use CRM_HRContactActionsMenu_Component_GroupSeparatorItem as GroupSeparatorItem;
+use CRM_HRCore_Service_Manager as ManagerService;
+use CRM_HRCore_Component_ContactActionsMenu_NoSelectedLineManagerTextItem as NoSelectedLineManagerTextItem;
+use CRM_HRCore_Component_ContactActionsMenu_LineManagersListItem as LineManagersListItem;
+
+/**
+ * Class CRM_HRCore_Helper_ContactActionsMenu_WorkflowActionGroup
+ */
+class CRM_HRCore_Helper_ContactActionsMenu_WorkflowActionGroup {
+
+  /**
+   * @var ManagerService
+   */
+  private $managerService;
+
+  /**
+   * @var array
+   */
+  private $contactID;
+
+  /**
+   * @var array
+   */
+  private $workflowCaseTypes;
+
+  /**
+   * CRM_HRCore_Helper_ContactActionsMenu_WorkflowActionGroup constructor.
+   *
+   * @param ManagerService $managerService
+   * @param int $contactID
+   */
+  public function __construct(ManagerService $managerService, $contactID) {
+    $this->managerService = $managerService;
+    $this->contactID = $contactID;
+  }
+
+  /**
+   * Gets Workflow Menu Group with menu items already
+   * added.
+   */
+  public function get() {
+    $actionsGroup = new ActionsGroup('Workflows:');
+    $actionsGroup->addItem($this->getWorkflowJoiningButton());
+    $actionsGroup->addItem($this->getWorkflowExitingButton());
+    $actionsGroup->addItem($this->getWorkflowOtherButton());
+    $actionsGroup->addItem(new GroupSeparatorItem());
+    $actionsGroup->addItem($this->getWorkflowNewTaskButton());
+    $actionsGroup->addItem($this->getWorkflowNewDocumentButton());
+    $actionsGroup->addItem(new GroupSeparatorItem());
+
+    $lineManagers = $this->getLineManagers();
+
+    if ($lineManagers) {
+      $lineManagersListItem = new LineManagersListItem($this->managerService, $this->contactID);
+      $actionsGroup->addItem($lineManagersListItem);
+      $actionsGroup->addItem($this->getManageLineManagerButton());
+    }
+    else {
+      $noLineManagerTextItem = new NoSelectedLineManagerTextItem();
+      $actionsGroup->addItem($noLineManagerTextItem);
+      $actionsGroup->addItem($this->getAddLineManagerButton());
+    }
+
+    return $actionsGroup;
+  }
+
+  /**
+   * Gets the Workflow Joining button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowJoiningButton() {
```

Comment:
The class name already says this is the Workflow Group, so I think we don't need to mention that on the methods as well. We can just name them `getJoiningButton()`, `getExitingButton()` and so on